src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentationMeterProviderBuilderExtensions.cs

```diff
@@ -27,6 +27,11 @@ public static class AspNetCoreInstrumentationMeterProviderBuilderExtensions
 #if NET8_0_OR_GREATER
         return builder.ConfigureMeters();
 #else
+        if (Environment.Version.Major >= 8)
```

Comment:
I would rather propose removing all pre-processing directives like `NET8_0_OR_GREATER` and using the equivalent of `Environment.Version.Major` instead of introducing a new condition which does the same operation.

Reply:
I was considering it. It introduces additional checks in runtime. Mostly on .NET 8. I have tried to avoid this additional _overhead_, even if it is minimal.

What is more, .NET6/.NET7 branches should be removed in November.

Reply:
Done in 3054ec1ed2273e3acdeed59be598b464b9d3d2db